app/controllers/ops_controller/settings/cap_and_u.rb

```diff
@@ -128,21 +124,18 @@ def cu_build_edit_screen
       end
     end
     if @edit[:current][:clusters].present?
-      @cluster_tree = TreeBuilderClusters.new(:cluster_tree,
-                                              @sb,
-                                              true,
-                                              :root => @edit[:current])
+      @cluster_tree = TreeBuilderClusters.new(:cluster_tree, @sb, true)
     end
-    @edit[:current][:storages] = []
-    @st_recs = {}
-    Storage.in_my_region.includes(:taggings, :tags, :hosts).select(:id, :name, :store_type, :location)
-           .sort_by { |s| s.name.downcase }.each do |s|
-      @st_recs[s.id] = s
-      @edit[:current][:storages].push(:name       => s.name,
-                                      :id         => s.id,
-                                      :capture    => s.perf_capture_enabled?,
-                                      :store_type => s.store_type,
-                                      :location   => s.location) # fields we need
+    @edit[:current][:storages] = {}
+    Storage.in_my_region.includes(:taggings, :tags, :hosts).select(:id, :name, :store_type, :location).sort_by { |s| s.name.downcase }.each do |s|
+      @edit[:current][:storages][s.id] = {
+        :name       => s.name,
+        :id         => s.id,
+        :capture    => s.perf_capture_enabled?,
+        :st_rec     => s,
```

Comment:
saving an active record object in a session?

Reply:
Well, there's more that I didn't add here. Should I clean it up with the rest?

Reply:
Well, we've had problems before that were caused by AR in session,
so, I think we should not introduce it here..

Reply:
@skateman I thought we were no longer storing ARObjects in a session, we should only store their ids

Reply:
Cleaning it up... :scissors: